frontend/src/app/app.component.ts

```diff
@@ -1,31 +1,109 @@
-import {Component} from '@angular/core';
-import {ApiService} from './api.service';
-import {SegmentedTimelineDataResponse, SegmentData, SegmentedTimelineCompareResponse, SegmentedDataErrorResponse} from './proto/protobuf/data';
-import {Subject} from 'rxjs';
+import { Component, Injectable } from '@angular/core';
+import { ApiService } from './api.service';
+import {
+  SegmentedTimelineDataResponse,
+  SegmentData
+} from './proto/protobuf/data';
+import { Subject } from 'rxjs';
+import {
+  NgbModal,
+  NgbDateStruct,
+  NgbDateAdapter,
+} from '@ng-bootstrap/ng-bootstrap';
+
+/**
+ * This Service handles how the date is represented in scripts i.e. ngModel.
+ */
+@Injectable()
+export class CustomAdapter extends NgbDateAdapter<string> {
+  readonly DELIMITER = '-';
+
+  fromModel(value: string | null): NgbDateStruct | null {
+    if (value) {
+      const date = value.split(this.DELIMITER);
+      return {
+        day: parseInt(date[1], 10),
+        month: parseInt(date[0], 10),
+        year: parseInt(date[2], 10),
+      };
+    }
+    return null;
+  }
+
+  toModel(date: NgbDateStruct | null): string | null {
+    return date
+      ? date.month + this.DELIMITER + date.day + this.DELIMITER + date.year
+      : null;
+  }
+}
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css']
+  styleUrls: ['./app.component.css'],
+  providers: [{ provide: NgbDateAdapter, useClass: CustomAdapter }],
 })
 export class AppComponent {
   title = 'Inventory Visualizer';
   dataResponse: SegmentedTimelineDataResponse;
   data: Array<any[]>;
   segments: SegmentData[];
   page: number;
+
   updateCharts: Subject<any> = new Subject();
+  clearCharts: Subject<any> = new Subject();
+  updateFilters: Subject<any> = new Subject();
+
   files: string[];
   hasDisplayedData: boolean;
 
+  filtered: boolean;
+  filters: {
+    device: string;
+    countries: string[];
+    fromDate: string;
+    toDate: string;
+  };
+  fromDate: string;
+  toDate: string;
+  countries = [];
+  devices = [
+    { id: 'MT', name: 'Mobile' },
+    { id: 'DO', name: 'Desktop' },
```

Comment:
Populating dynamically is preferable to hardcoding values here. Is there an implementational reason for devices being treated differently than countries?

Reply:
I had assumed that this would not be changing but I guess I should change that.

Reply:
done